### PR TITLE
[1.14] Flattening

### PR DESCRIPTION
**Pull request by temp1011**
Mostly cleanup of broken code through porting. Farm block textures/models may be more broken now but shouldn't be too bad to fix.

### CODE REVIEW

File: src/main/java/forestry/farming/ModuleFarming.java [hunk 1]
```diff
@@ -242,28 +242,28 @@ public void doInit() {
 
 		Circuits.farmArborealManaged = new CircuitFarmLogic("managedArboreal", arborealFarm, false);
 		Circuits.farmArborealManual = new CircuitFarmLogic("manualArboreal", arborealFarm, true);
-		arborealFarm.registerSoil(new ItemStack(Blocks.DIRT), coreBlocks.humus.getDefaultState());
-		arborealFarm.registerSoil(new ItemStack(coreBlocks.humus), coreBlocks.humus.getDefaultState());
+//		arborealFarm.registerSoil(Blocks.DIRT);	//TODO check this doesn't allow players to use dirt as soil in managed farm
```
Comment (temp1011) [inline src/main/java/forestry/farming/ModuleFarming.java]: I'm not really sure how this worked before. Perhaps I can't simplify the API here

Comment (Nedelosk) [reply src/main/java/forestry/farming/ModuleFarming.java]: I think it makes more sense to just keep `registerSoil(ItemStack, BlockState)` so other mods could register some sort of item and place a block that is completely unconnected to the item.

Comment (temp1011) [reply src/main/java/forestry/farming/ModuleFarming.java]: Ok, I've kept the old API

**Comment by temp1011** [comment]
Blocked on resolution of https://github.com/MinecraftForge/MinecraftForge/issues/6177 so that the minecarts are fixed.

**Pull request merged**